pkg/azureutils/azure_disk_utils.go

```diff
@@ -900,12 +901,20 @@ func GetAzVolumeAttachmentState(volumeAttachmentStatus storagev1.VolumeAttachmen
 
 type UpdateCRIFunc func(client.Object) error
 
-func UpdateCRIWithRetry(ctx context.Context, informerFactory azdiskinformers.SharedInformerFactory, cachedClient client.Client, azDiskClient azdisk.Interface, obj client.Object, updateFunc UpdateCRIFunc, maxNetRetry int, updateMode CRIUpdateMode) error {
+func UpdateCRIWithRetry(ctx context.Context, informerFactory azdiskinformers.SharedInformerFactory, cachedClient client.Client, azDiskClient azdisk.Interface, obj client.Object, updateFunc UpdateCRIFunc, maxNetRetry int, updateMode CRIUpdateMode) (client.Object, error) {
 	var err error
 	objName := obj.GetName()
-	ctx, w := workflow.New(ctx)
+	var callerName string
+	pc, _, _, ok := runtime.Caller(2)
+	if ok {
+		details := runtime.FuncForPC(pc)
+		callerName = details.Name()
+	}
+	ctx, w := workflow.New(ctx, workflow.WithDetails("caller", callerName))
```

Comment:
Consider adding a `workflow.WithCaller(skip int)` option function that would get and set the `"caller"` detail so this could be reused elsewhere.